le_lambda.py

```diff
@@ -147,31 +132,65 @@ def lambda_handler(event, context):
                 cloud_trail = json.loads(data)
                 for event in cloud_trail['Records']:
-                    s.sendall('%s %s\n' % (log_token, json.dumps(event)))
+                    send_to_r7(json.dumps(event), sock, LOG_TOKEN)
+                logger.info('Finished sending file={} to R7'.format(key))
             else:
+                logger.info('File={} is unrecognized log format. Sending raw lines to R7'.format(key))
                 for line in lines:
-                    s.sendall('%s %s\n' % (log_token, line))
-            for token in tokens:
-                s.sendall('%s %s\n' % (token, "username='{}' finished sending log data end_time='{}'"
-                                       .format(username, str(datetime.datetime.utcnow()))))
+                    send_to_r7(line, sock, LOG_TOKEN)
+                logger.info('Finished sending file={} to R7'.format(key))
         except Exception as e:
-            for token in tokens:
-                print e
-                s.sendall('%s %s\n' % (token, "Error getting username='{}' file='{}' from bucket='{}'. Make sure "
-                                              "they exist and your bucket is in the same region as this function."
-                                       .format(username, key, bucket)))
+            logger.error('Error getting file={} Check permissions and verify this function '
+                         'resides in the same region as the S3 bucket={}'.format(key, bucket))
+            logger.error(e)
         finally:
-            s.close()
+            sock.close()
+            logger.info('Function execution finished.')
+
+
+def create_socket():
+    logger.info('Creating SSL socket')
+    s_ = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s = ssl.wrap_socket(
+        sock=s_,
+        keyfile=None,
+        certfile=None,
+        server_side=False,
+        cert_reqs=ssl.CERT_REQUIRED,
+        ssl_version=getattr(
+            ssl,
+            'PROTOCOL_TLSv1_2',
+            ssl.PROTOCOL_TLSv1
+        ),
+        ca_certs=certifi.where(),
+        do_handshake_on_connect=True,
+        suppress_ragged_eofs=True,
+    )
+    try:
+        logger.info('Connecting to {}:{}'.format(ENDPOINT, PORT))
+        s.connect((ENDPOINT, PORT))
+        return s
+    except socket.error, exc:
+        logger.error('Exception socket.error : {}'.format(exc))
+
+
+def send_to_r7(line, r7_socket, token):
+    r7_socket.sendall('{} {}\n'.format(token, line))
 
 
 def validate_uuid(uuid_string):
-    regex = re.compile('^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$', re.I)
-    match = regex.match(uuid_string)
-    return bool(match)
+    try:
+        val = UUID(uuid_string)
+    except Exception as uuid_exc:
+        logger.error('Can not validate token: ' + uuid_exc)
+        return False
+    return val.hex == uuid_string.replace('-', '')
```

Comment:
an exception will be thrown by uuid module in any invalid uuid so I don't think you need it